### PR TITLE
Accessibility Project - Pre-assessments

### PR DESCRIPTION
**Pull request by SebastianBurke**
The WCAG 2.1 pre-assessments included in this pull request are as follows:

GC Top Menu
GC Topic
Header Language Selector
Tabs Interference Carousel

### CODE REVIEW

File: common/gc_top_menu/reports/pre-a11y-1.json [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
+    {
+      "earl:test": "WCAG21:bypass-blocks",
+      "earl:subject": "_:subject",
+      "earl:outcome": "earl:untested",
+      "dct:description": "",
+      "earl:mode": "earl:unknownMode",
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
```
Comment (duboisp) [inline common/gc_top_menu/reports/pre-a11y-1.json:276]: Note: This test is usually only required when testing the whole page and not part of it. The bypass block is the first interactive element the user will found when navigating on a webpage. You can see those link when tabbing toward the web page content from the browser navigation bar. The skip link should be the first web content item.

The outcome should be inapplicable in the context of only testing the top menu.

File: common/gc_top_menu/reports/pre-a11y-1.json [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+    {
+      "earl:test": "WCAG21:page-titled",
+      "earl:subject": "_:subject",
+      "earl:outcome": "earl:untested",
+      "dct:description": "",
+      "earl:mode": "earl:unknownMode",
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
```
Comment (duboisp) [inline common/gc_top_menu/reports/pre-a11y-1.json:284]: Again, this is only applicable when testing the whole page.

The scope of this test is to ensure the text content of the `<title>` element is meaningful to the web page content and unique. It also need to be aligned with the header level 1 of the page.

File: common/gc_top_menu/reports/pre-a11y-1.json [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+    {
+      "earl:test": "WCAG21:language-of-page",
+      "earl:subject": "_:subject",
+      "earl:outcome": "earl:untested",
+      "dct:description": "",
+      "earl:mode": "earl:unknownMode",
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
```
Comment (duboisp) [inline common/gc_top_menu/reports/pre-a11y-1.json:364]: This is only applicable when testing the whole page, not part of it. 

This success criteria is about to test the `lang` attribute set on the `<html>` element

File: common/gc_top_menu/reports/pre-a11y-1.json [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+    {
+      "earl:test": "WCAG21:error-identification",
+      "earl:subject": "_:subject",
+      "earl:outcome": "earl:untested",
+      "dct:description": "",
+      "earl:mode": "earl:unknownMode",
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
```
Comment (duboisp) [inline common/gc_top_menu/reports/pre-a11y-1.json:412]: Can the gc top menu trigger an input error made by the user?

File: common/gc_top_menu/reports/pre-a11y-1.json [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+    {
+      "earl:test": "WCAG21:status-messages",
+      "earl:subject": "_:subject",
+      "earl:outcome": "earl:inapplicable",
+      "dct:description": "",
+      "earl:mode": "earl:unknownMode",
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    }
+  ]
+}
```
Comment (duboisp) [inline common/gc_top_menu/reports/pre-a11y-1.json:462]: Every content file in this repository must end with an empty line. The red circle with a dash symbol illustrate that issue and it is also reported as an error by our lint spacing continuous integration test. You can also run that test locally but executing the CLI command `grunt test`

```suggestion
}

```

File: common/gc_topic/reports/pre-a11y-1.json
Comment (duboisp) [inline common/gc_topic/reports/pre-a11y-1.json:1]: This file need to be moved in this folder: /templates/topic/

File: common/gc_topic/reports/pre-a11y-1.json [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+    "@id": "_:subject",
+    "dct:references": "https://wet-boew.github.io/GCWeb/templates/topic/topic-en.html",
+    "@type": [
+      "earl:TestSubject",
+      "schema:WebPage"
+    ],
+    "dct:description": "Evaluation of effectiveness of Topic Title at appropiately describing the content of the page",
+    "earl:pointer": {
+      "@type": "oa:CssSelector",
+      "@value": ""
```
Comment (duboisp) [inline common/gc_topic/reports/pre-a11y-1.json:16]: For the topic page, has you pre-assessed every HTML element or only the one contained in the `<main>` element?
Was there some sub-element that was excluded from the pre-assessment which would means that the assessment is limited to the integration of those component inside this template.

File: common/gc_topic/reports/pre-a11y-1.json [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
+    {
+      "earl:test": "WCAG21:audio-description-or-media-alternative-prerecorded",
+      "earl:subject": "_:subject",
+      "earl:outcome": "earl:untested",
+      "dct:description": "",
+      "earl:mode": "earl:unknownMode",
+      "@type": ["earl:TestResult", "earl:Assertion"]
+    },
```
Comment (duboisp) [inline common/gc_topic/reports/pre-a11y-1.json:74]: Is this test applicable?

File: common/header_language_selector/reports/pre-a11y-1.json
Comment (duboisp) [inline common/header_language_selector/reports/pre-a11y-1.json:1]: This report need to be moved here: /sites/language/

File: common/header_language_selector/reports/pre-a11y-1.json [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+    "@id": "_:subject",
+    "dct:references": "https://wet-boew.github.io/GCWeb/sites/language/language-en.html",
+    "@type": [
+      "earl:TestSubject",
+      "schema:WebPage"
+    ],
+    "dct:description": "Evaluation of how the Language Toggle Link meets accessibility requirements",
+    "earl:pointer": {
+      "@type": "oa:CssSelector",
+      "@value": ""
```
Comment (duboisp) [inline common/header_language_selector/reports/pre-a11y-1.json:16]: A CSS selector need to be defined.

File: common/tabs_interference_carousel/reports/pre-a11y-1.json [hunk 9]
```diff
@@ -0,0 +1,444 @@
+{
+  "@context": "https://wet-boew.github.io/vocab/context/2023/conformance-report-en.json",
+
+  "@type": [ "earl:Assertion", "acr:AssessmentReport" ],
+
+  "earl:subject": {
+    "@id": "_:subject",
+    "dct:references": "https://wet-boew.github.io/GCWeb/sites/language/language-en.html",
```
Comment (duboisp) [inline common/tabs_interference_carousel/reports/pre-a11y-1.json:8]: Probably this is URL you means
https://wet-boew.github.io/gcweb-compiled-demos/wetboew-demos/tabs/tabs-carousel-en.html (which seem broken)
or https://wet-boew.github.io/wet-boew/demos/tabs/tabs-carousel-en.html

File: common/tabs_interference_carousel/reports/pre-a11y-1.json [hunk 10]
```diff
@@ -0,0 +1,444 @@
...
+    "@type": [
+      "earl:TestSubject",
+      "schema:WebPage"
+    ],
+    "dct:description": "Evaluation of how the Tabs Interface - Carousel component meets accessibility requirements",
+    "earl:pointer": {
+      "@type": "oa:CssSelector",
+      "@value": ""
```
Comment (duboisp) [inline common/tabs_interference_carousel/reports/pre-a11y-1.json:16]: Missing the CSS selector.

Please make it also specific to the carrousel and its variants

File: common/tabs_interference_carousel/index.json-ld
Comment (duboisp) [inline common/tabs_interference_carousel/index.json-ld:1]: We can keep this one in the common folder, but rename it for common/wb-tabs

File: common/tabs_interference_carousel/index.json-ld [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+	"title": {
+		"en": "Tabs Interface - Carousel",
+		"fr": "Interface à onglets - Carrousel"
+	},
+	"description": {
+		"en": "Responsive carousels are in-page components used to display multiple different items within the same content space.",
+		"fr": "Les carrousels responsifs sont des composants intégrés à la page utilisés pour afficher plusieurs éléments différents dans le même espace de contenu."
+	},
+	"modified": "2023-09-08",
+	"componentName": "Tabs Interface - Carousel",
```
Comment (duboisp) [inline common/tabs_interference_carousel/index.json-ld:18]: The componentName value must match the folder name.

**Review by duboisp — CHANGES_REQUESTED** [review]
See the comments, some of them is applicable on multiple files.

File: common/gc_topic/reports/pre-a11y-1.json (hunk 6 above)
Comment (SebastianBurke) [reply common/gc_topic/reports/pre-a11y-1.json:16]: Hi Pierre, I am not sure what this means. Which HTML elements  should I be pre-assessing for the topic page?

File: common/header_language_selector/reports/pre-a11y-1.json (hunk 8 above)
Comment (SebastianBurke) [reply common/header_language_selector/reports/pre-a11y-1.json:16]: I am unsure what css selector could be defined here as the language selector works as a link <a> with an href of "/fr.html"

File: common/gc_top_menu/reports/pre-a11y-1.json [hunk 12]
```diff
@@ -0,0 +1,462 @@
...
+      }
+    ],
+    "@type": [
+      "earl:TestSubject",
+      "schema:WebPage"
+    ],
+    "dct:description": "Describe what is being assessed, like the full page or a component inside that page. Provide a pointer when applicable (earl:pointer)",
+    "earl:pointer": {
+      "@type": "oa:CssSelector",
+      "@value": "title, meta, #wb-bc, main"
```
[the remark this answers is not in the document]
Comment (SebastianBurke) [reply common/gc_top_menu/reports/pre-a11y-1.json:34]: For topic title, what should the css selector value be? I couldn't find one

**Pull request closed without merging**